src/libraries/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs

```diff
@@ -89,13 +89,15 @@ static XsltApiTestCaseBase()
             {
                 Directory.CreateDirectory(Path.GetDirectoryName(s_temporaryResolverDocumentFullName));
 
-                // Replace absolute URI in xmlResolver_document_function.xml based on the environment
+                // Replace absolute URI in xmlResolver_document_function.xml based on the environment, and store it under a different name
                 string xslFile = Path.Combine("TestFiles", FilePathUtil.GetTestDataPath(), "XsltApi", "xmlResolver_document_function_absolute_uri.xsl");
+                string replacedXslFile = Path.Combine("TestFiles", FilePathUtil.GetTestDataPath(), "XsltApi", "xmlResolver_document_function_absolute_uri_replaced.xsl");
+                File.Copy(xslFile, replacedXslFile, true);
```

Comment:
I suspect infra folks might have encountered this kind of issue before @safern @ViktorHofer is there any other way to fix this issue instead of creating duplicate file?
Tests run successful on first build test, as the test file content updated with a random path rerunning the test without rebuilding fails

Reply:
I think this is the correct way to go, but instead, we should use the APIs in `FileCleanupTestBase` which uses the temp path and then on dispose cleans this up. That way we avoid collisions and errors where there is a handle using the file and you can later no longer copy it causing test failures on re-run.

https://github.com/dotnet/runtime/blob/696e72b14d29979d7dab11eb4384ae656fdfaba6/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/IO/FileCleanupTestBase.cs#L83

That will give you a file path in the temp directory which contains the test name that created it, if debugging is needed. 

Would that work here?

Reply:
I have actually tried with reverting back the content on dispose but this class was disposed before the tests run so tests were failing, I assume if we clean up copied file on dispose same would happen ...

Reply:
I'm not sure if there's a way to call Dispose at the "right time". xUnit creates an instance per each test case (presumably so it parallelises better), and the static fields are shared across the instances. This means we can't dispose of the data in as part of the test cleanup (because they're shared).

I suppose we could copy & replace the content using `FileCleanupTestBase` in individual tests, but it... somewhat seemed unnecessary to make 10+ copies of the same content.

Reply:
Yeah, makes sense. That’s why I mentioned that the static constructor might make it impossible to do it.

Let’s then leave it as is as I can’t think of a better way.